ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py

```diff
@@ -0,0 +1,84 @@
+# Testing Function
+# global
+from hypothesis import strategies as st
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+
+
+# from ivy_tests.test_ivy.test_functional.test_core.test_statistical import (
+#   statistical_dtype_values,
+# )
+import ivy_tests.test_ivy.test_frontends.test_numpy.helpers as np_frontend_helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+
+# import ivy
+
+
+@handle_frontend_test(
+    fn_tree="numpy.diagonal",
+    # dtype_x_axis=helpers.dtype_values_axis(
+    #   available_dtypes=helpers.get_dtypes("numeric"),
+    #  num_arrays=1,
+    # shared_dtype = True,
+    # min_num_dims = 2
+    # ),
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=1,
+        # min_value=0,
+        # max_value = 1,
+        shared_dtype=True,
+        min_num_dims=2,
+    ),
+    # where=np_frontend_helpers.where(),
+    # dtype_and_axis1=helpers.dtype_values_axis(
+    #   available_dtypes=helpers.get_dtypes("numeric"),
+    # num_arrays=1,
+    # min_value=0,
+    # max_value = 1,
+    #  shared_dtype=True,
+    # min_num_dims=2,
+    # ),
+    offset=st.integers(min_value=-1, max_value=1),
+    axis1=st.integers(min_value=-2, max_value=2),
+    axis2=st.integers(min_value=-2, max_value=2),
+    # axis1!=axis2,
+    # dtype=helpers.get_dtypes("float", full=False, none=True),
```

Comment:
You could use this [axis helper](https://github.com/unifyai/ivy/blob/c3bc6bb192c1e12f666ea3a9c66428c2062c6f4e/ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py#L289) that has the `unique` argument built in. This will eliminate the need for assertions.

Reply:
I used it now by giving a random shape. Still giving assertion error due to same axis. I called it twice for each axis. but may be I have to define according to "hypothesis strategy from which the shape will be drawn".

Reply:
But, now that I think about it more, you could use [`dtype_values_axis`](https://github.com/unifyai/ivy/blob/697353a0f7b23f1e204a9399ec0f831b299aa0be/ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py#L240) that kind of combines `dtype_and_values` and `get_axis`. In that way you won't have to worry about the shape.

Reply:
> Yes, you get the error because you call it twice. You should call it only once with `min_size=2` and `max_size=2`, and then just get axis1 from position 0 and axis2 from position 1 of the axis return.

axis is an int object. I didn't understand your point. I just use if-else condition to create two different values of axis.
Now it is showing passed, :).

Reply:
`dtype_values_and_axis` returns an int, that's right. But `get_axis` returns a tuple, i.e. more than one axes, which you can then access by position (axis1 = axes[0], axis2 = axes[1]. That's what I meant.

I've actually found an example that will help you. You can mimic what is being done [here ](https://github.com/unifyai/ivy/blob/58714a8f1ecd6cc3e06551b02f7230db41e553c1/ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_transpose_like_operations.py#L81). Only difference would be that you want to set `unique=True` in the `get_axis` call.